CriteoPublisherSdk/Sources/Logging/CR_RemoteLogRecordSerializer.m

```diff
@@ -0,0 +1,68 @@
+//
+//  CR_RemoteLogRecordSerializer.m
+//  CriteoPublisherSdk
+//
+//  Copyright © 2018-2021 Criteo. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#import "CR_RemoteLogRecordSerializer.h"
+
+@implementation CR_RemoteLogRecordSerializer
+
+- (NSArray<NSDictionary<NSString *, NSObject *> *> *)serializeRecords:
+    (NSArray<CR_RemoteLogRecord *> *)records {
+  NSMutableArray *formattedArray = [NSMutableArray arrayWithCapacity:records.count];
+  for (CR_RemoteLogRecord *record in records) {
+    NSDictionary<NSString *, NSObject *> *dictionary = [self serializeRecord:record];
+    [formattedArray addObject:dictionary];
+  }
+  return formattedArray;
+}
+
+- (NSDictionary<NSString *, NSObject *> *)serializeRecord:(CR_RemoteLogRecord *)record {
+  // The exception type is nullable, so it is not possible to use @{} notation.
```

Comment:
NIT: this is something I'm not asking you to do, but you could init that mutable dict with a literal ;)

Reply:
could you show me the syntax ?